docs/source/task_tutorial.rst

```diff
@@ -0,0 +1,216 @@
+        
+        # set up paths to data (specific to each dataset)
+        dt = opt['datatype'].split(':')[0]
+        labels_path = os.path.join(opt['datapath'], 'mnist', dt, 'labels.json')
+        image_path = os.path.join(opt['datapath'], 'mnist', dt)
+        return labels_path, image_path
+
+By creating ``MnistQATeacher`` as a subclass of ``DialogTeacher``, the job of creating a teacher for this task becomes much simpler: most of the work that needs to be done will limit itself to defining a ``setup_data`` method. This method is a generator that will take in a path to the data and yield a pair of elements for each call. The first element of the pair is a tuple containing the following information: ``(query, labels, reward, label_candidates, path_to_image)``. The second is a boolean flag ``episode_done?`` which indicates if the current query marks the end of an episode or not. 
+
+More information on this format can be found in the documentation on ``data_loader`` in `DialogueData <http://parl.ai/static/docs/dialog.html#parlai.core.dialog_teacher.DialogData>`__ (``setup_data`` is provided as a data_loader to ``DialogueData``).
+
+The sample ``setup_data`` method for our task is presented below. 
+
+.. code-block:: python
+
+    def setup_data(self, path):
+        print('loading: ' + path)
+
+        # open data file with labels 
+        # (path will be provided to setup_data from opt['datafile'] defined above)
+        with open(path) as labels_file:
+            self.labels = json.load(labels_file)
+
+        # define standard question, since it doesn't change for this task
+        self.question = 'Which number is in the image?'
+        # every episode consists of only one query in this task
+        episode_done = True
+
+        # define iterator over all queries
+        for i in range(len(self.labels)):
+            # set up path to curent image
+            img_path = os.path.join(self.image_path, '%05d.bmp' % i)
+            # get current label, both as a digit and as a text
+            label = [self.labels[i], self.num_strs[int(self.labels[i])]]
+            # yield tuple with information and episode_done? flag
+            yield (self.question, label, None, None, img_path), episode_done
+
+As we can see from the code above, for this specific task the question is always the same, and thus it is fixed. For different tasks, this might change at each iteration. Similarly, for this task, each episode consist of only one query, thus ``episode_done?`` is always true (*i.e.*, each query is the end of its episode). This could also vary depending on the task.
+
+Looking at the tuple provided by the iterator at each yield, we can see that we defined a query, a label and an image path. When working with ``DialogTeacher`` in visual tasks, it is important to provide the path to the image in the ``setup_data`` tuple. This also allows for integrated processing (*e.g.*, conversion to ascii image by setting command line flag ``-im ascii``).
+
```

Comment:
"...``setup_data`` tuple. This allows one to inherit functionality around the "image-mode" command line parameter, such as automatically returning ascii versions of images if ``-im ascii`` is set.